aeneas/src/wasm/WasmTarget.v3

```diff
@@ -89,7 +89,15 @@ class WasmTarget extends Target {
 	def configureCompiler(compiler: Compiler) {
 		compiler.Reachability = true;
 		compiler.NormConfig.setSignatureLimits(10000, if(CLOptions.WASM_MULTI_VALUE.val, 1000, 1));
-		compiler.NormConfig.UnifyRepresentations = UnboxingUtil.makeMachUnifyRepresentations(getRuntime);
+		compiler.NormConfig.GetScalar = getScalar;
+	}
+	private def getScalar(compiler: Compiler, prog: Program, t: Type) -> Scalar.set {
+		match (t) {
+			x: BoolType => return Scalar.B32 | Scalar.B64;
+			x: IntType => return if (x.width <= 32, Scalar.B32 | Scalar.B64, Scalar.B64); // XXX: Scalar.R64, once packed refs
```

Comment:
You could write `Scalar.B64 | if(x.width <= 32, Scalar.B32)`

Reply:
This doesn't seem to work. I think what's happening is `if(x.width <= 32, Scalar.B32)` is typechecked as having type `Scalar` instead of `Scalar.set`, so its default value in the `false` case will be `Scalar.B32` too (instead of the empty set).